src/test/net_tests.cpp

```diff
@@ -319,4 +321,73 @@ BOOST_AUTO_TEST_CASE(PoissonNextSend)
     g_mock_deterministic_tests = false;
 }
 
+void message_serialize_deserialize_test(bool v2, const std::vector<CSerializedNetMsg>& test_msgs) {
+    // use 32 bytey keys with all zeros
+    CPrivKey k1(32, 0);
+    CPrivKey k2(32, 0);
+
+    // construct the serializers
+    std::unique_ptr<TransportSerializer> serializer;
+    std::unique_ptr<TransportDeserializer> deserializer;
+
+    if (v2) {
+        serializer = MakeUnique<V2TransportSerializer>(V2TransportSerializer(k1, k2));
+        deserializer = MakeUnique<V2TransportDeserializer>(V2TransportDeserializer(Params().MessageStart(), k1, k2));
+    }
+    else {
+        serializer = MakeUnique<V1TransportSerializer>(V1TransportSerializer());
+        deserializer = MakeUnique<V1TransportDeserializer>(V1TransportDeserializer(Params().MessageStart(), SER_NETWORK, INIT_PROTO_VERSION));
+    }
+    // run a couple of times through all messages with the same AEAD instance
+    for (unsigned int i=0; i<100;i++) {
+        for(const CSerializedNetMsg& msg_orig : test_msgs) {
+            // bypass the copy protection
+            CSerializedNetMsg msg;
+            msg.data = msg_orig.data;
+            msg.command = msg_orig.command;
+            size_t raw_msg_size = msg.data.size();
+
+            std::vector<unsigned char> serialized_header;
+            serializer->prepareForTransport(msg, serialized_header);
+
+            // read two times
+            //  first: read header
```

Comment:
Not sure if this was intended or not, but if it was intended I don't understand why
```suggestion
            // first: read header
```